scripts/update_function_docs/update_function_docs.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Usage: update_function_docs <RELEASE_BRANCH>
+//
+// e.g. update_function_docs origin/apply-setters/v0.2
+//
+// The command will checkout the release branch and update the function/example
+// docs with the latest patch version for the release. If the docs are updated
+// then a commit is created with the changes. The manual steps left to the user
+// are to push the commit to a branch and create a pull request.
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
+)
+
+func exitWithErr(err error) {
+	fmt.Fprintf(os.Stderr, "%v\n", err)
+	os.Exit(1)
+}
+
+func runCmd(name string, arg ...string) (string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command(name, arg...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	fmt.Printf("%s\n", cmd.String())
+	err := cmd.Run()
+	if err != nil {
+		return stdout.String(), fmt.Errorf("%s\n%s", stderr.String(), err)
```

Comment:
Why not print both stderr and stdout?
If we choose to do that, we can use https://pkg.go.dev/os/exec#Cmd.CombinedOutput

Reply:
I was fairly deliberate here with what I chose to print in order to provide useful yet not overly verbose output

The structure here is intended to print stderr in the case of a fatal error (i.e. exitWithError).

The stdout is deliberately printed for some of the commands to show to the user what changes were made (e.g. `gitCommit`, `gitShow`). It becomes a bit too verbose if the stdout is printed for all of the commands. Also, I think it is useful to keep stdout separate from stderr for the purpose of parsing stdout in the code.